src/bincode/bincode.zig

```diff
@@ -149,6 +149,15 @@ pub fn read(allocator: std.mem.Allocator, comptime U: type, reader: anytype, par
                     @field(data, field.name) = try bincode.read(allocator, field.type, reader, params);
                 }
             }
+
+            const init_field = "!bincode-config:init";
+            if (@hasDecl(T, init_field)) {
+                const field_config = @field(T, init_field);
+                if (field_config.init_fn) |init| {
+                    init(&data);
+                }
+            }
+
```

Comment:
This doesn't actually respect the added field in `FieldConfig`, making either one or the other redundant, or the former is missing code that actually uses it.

But beyond that, both this and the added field in `FieldConfig` is what I was wanting to avoid. It's this kind of functionality that should, in my view, be added to a field via a custom `serialize` function, leaving bincode logic simple.

Reply:
for more context - the approach ink's talking about is similar to [this](https://github.com/Syndica/sig/blob/20a6d3dd4a2f47ceea50ba2faad9e622a60ecc54/src/bincode/arraylist.zig#L17) - where after the serialization, you populate the cache in the same `serialize` function

Reply:
We spoke about this in a huddle, we'll proceed with this as is, but I have a plan to improve the situation, which is out of scope for this PR.